aws/resource_aws_lambda_function.go

```diff
@@ -665,6 +665,10 @@ func needsFunctionCodeUpdate(d resourceDiffer) bool {
 	return d.HasChange("filename") || d.HasChange("source_code_hash") || d.HasChange("s3_bucket") || d.HasChange("s3_key") || d.HasChange("s3_object_version")
 }
 
+func needsFunctionConfigUpdate(d resourceDiffer) bool {
+	return d.HasChange("description") || d.HasChange("handler") || d.HasChange("memory_size") || d.HasChange("role") || d.HasChange("timeout") || d.HasChange("kms_key_arn") || d.HasChange("layers") || d.HasChange("dead_letter_config") || d.HasChange("tracing_config") || d.HasChange("vpc_config") || d.HasChange("runtime") || d.HasChange("environment")
```

Comment:
Downside to this PR is checking each of these fields twice worst case, as the existing logic still checks each individual field, although this part short-circuits in most cases.